### PR TITLE
Fix cli share_workspace

### LINKED ISSUE #1682: CLI share_workspace broken
The CLI share_workspace command does not provide a way to configure the user role 

```
Usage: parsec core share_workspace [OPTIONS] WORKSPACE_NAME USER_ID

  Create a new workspace for the given device.

Options:
  --config-dir DIRECTORY
  -l, --log-level [DEBUG|INFO|WARNING|ERROR]
  -f, --log-format [CONSOLE|JSON]
  -o, --log-file TEXT
  --log-filter TEXT
  -D, --device TEXT               Device to use designed by it ID, see
                                  `list_devices` command to get the available
                                  IDs  [required]
  -P, --password TEXT
  --help                          Show this message and exit.
```

The command is unusable:
```
$ parsec core share_workspace -P test -D 806 alice_workspace 18e
Error: workspace_share() missing 1 required positional argument: 'role'

```

### PR DESCRIPTION
**Pull request by AntoineDupre**
closes  #1682

### CODE REVIEW

File: parsec/core/cli/share_workspace.py [hunk 1]
```diff
@@ -7,20 +7,35 @@
...
+        await core.user_fs.workspace_share(f"/{name}", user_id, role)
 
 
 @click.command(short_help="share workspace")
 @core_config_and_device_options
 @click.argument("workspace_name")
 @click.argument("user_id", type=UserID, required=True)
-def share_workspace(config, device, workspace_name, user_id, **kwargs):
+@click.argument(
+    "user_role", default="READER", type=click.Choice(("READER", "CONTRIBUTOR", "MANAGER", "OWNER"))
```
[the remark this answers is not in the document]
Comment (touilleMan) [reply parsec/core/cli/share_workspace.py:34]: Aaaaaand while I'm at it: how do we know what to provide for user_id ?
Given user_id is now mainly an uuid, it's hard for the end user to know about it (there is no cli command to retreive this uuid from the human handle for instance)

The solutions I see:
1) provide a `--recipiant` option that search on the human handle
2) create a new `parsec core human_find` command to retrieve a user_id from human handle
3) do both (so you can choose between `share_workspace [...] --recipiant=john@doe.com` or `share_workspace [...] --user-id=415e9218e43845d388694b734141a288`, typically if you're working on a non-human user)

If we choose to do changes on this, I think we should do that in another PR ;-)

[the remark this answers is not in the document]
Comment (touilleMan) [reply parsec/core/cli/share_workspace.py:34]: Big realization: There is now way to pass the `None` value !!!!! so we cannot remove a role...

File: parsec/core/cli/share_workspace.py [hunk 2]
```diff
@@ -7,20 +7,35 @@
 from parsec.cli_utils import cli_exception_handler
 from parsec.core import logged_core_factory
 from parsec.core.cli.utils import core_config_and_device_options
+from parsec.core.types import WorkspaceRole
 
 
-async def _share_workspace(config, device, name, user_id):
+def cli_role_to_workspacerole(role):
+    roles = {
```
[the remark this answers is not in the document]
Comment (touilleMan) [reply parsec/core/cli/share_workspace.py:14]: As a matter of fact (after realizing we also need the `None`, see below), I think we should do all the conversion in the click argument:
```python

class WorkspaceRoleChoice(click.Choice):
    STR_TO_ROLE = {
        "NONE": None,
        *{role.value: role for role in WorkspaceRole}
    }

    def __init__(self, **kwargs):
        super().__init__(**kwargs)

    def convert(self, value, param, ctx):
        ret = super().convert(value, param, ctx)
        return self.STR_TO_ROLE[ret.upper()]

@click.argument(
    "--role", type=WorkspaceRoleChoice(case_sensitive=False)
)
[...]
```

**Review by touilleMan — CHANGES_REQUESTED** [review]

File: parsec/core/cli/share_workspace.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (AntoineDupre) [reply parsec/core/cli/share_workspace.py]: I created a new issue about retriving the human handle from cli ##1712


The lack of `None` value in click is a bit problematic. I setup a conversion method: 
```python

    @staticmethod
    def fix_none_choice(choice):
        return None if choice == "None" else choice
```

File: parsec/core/cli/share_workspace.py [hunk 3]
```diff
@@ -7,20 +7,35 @@
...
+        "OWNER": WorkspaceRole.OWNER,
+    }
+    return roles[role]
+
+
+async def _share_workspace(config, device, name, user_id, user_role):
     async with logged_core_factory(config, device) as core:
-        await core.user_fs.workspace_share(f"/{name}", user_id)
+        role = cli_role_to_workspacerole(user_role)
+        await core.user_fs.workspace_share(f"/{name}", user_id, role)
```
[the remark this answers is not in the document]
Comment (AntoineDupre) [reply parsec/core/cli/share_workspace.py]: Oh yes ! it has to be converted. I put a helper in the `LoggedCore` to find the WorkspaceEntry from a name:
```python
    def find_workspace_from_name(self, workspace_name: str):
        for workspace in self.user_fs.get_user_manifest().workspaces:
            if workspace_name == workspace.name:
                return workspace
        raise FSWorkspaceNotFoundError(f"Unknown workspace {workspace_name}")
```

File: parsec/core/cli/share_workspace.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (AntoineDupre) [reply parsec/core/cli/share_workspace.py]: :+1:

File: parsec/core/cli/share_workspace.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (touilleMan) [reply parsec/core/cli/share_workspace.py]: I think we'd better use the workspace-id directly (especially given we can have multiple workspaces with the same name !)

But this is "fine enough" for now, we should merge this PR as is and open an issue (especially given we will probably need a way to list the user's workspaces and it ids with this change !)

**Review by touilleMan — APPROVED** [review]

**Pull request merged**